pkg/security/security_profile/manager.go

```diff
@@ -0,0 +1,2376 @@
+			return model.WorkloadWarmup
+		}
+		return model.ProfileAtMaxSize
+	}
+	return profileState
+}
+
+func (m *Manager) incrementEventFilteringStat(eventType model.EventType, state model.EventFilteringProfileState, result EventFilteringResult) {
+	m.eventFiltering[eventFilteringEntry{eventType, state, result}].Inc()
+}
+
+// CountEvictedVersion count the evicted version for associated metric
+func (m *Manager) countEvictedVersion(imageName, imageTag string) {
+	m.evictedVersionsLock.Lock()
+	defer m.evictedVersionsLock.Unlock()
+	m.evictedVersions = append(m.evictedVersions, cgroupModel.WorkloadSelector{
+		Image: imageName,
+		Tag:   imageTag,
+	})
+}
+
+func (m *Manager) persist(p *profile.Profile, formatsRequests map[config.StorageFormat][]config.StorageRequest) error {
+	for format, requests := range formatsRequests {
+		p.Metadata.Serialization = format.String()
+
+		// encode profile
+		data, err := p.Encode(format)
+		if err != nil {
+			seclog.Errorf("couldn't encode profile [%s] to %s format: %v", p.GetSelectorStr(), format, err)
+			continue
+		}
+
+		for _, request := range requests {
+			var storage ActivityDumpStorage
+			switch request.Type {
+			case config.LocalStorage:
+				storage = m.localStorage
+			case config.RemoteStorage:
+				storage = m.remoteStorage
+			default:
+				seclog.Errorf("couldn't persist [%s] to %s format: unknown storage type: %s", p.GetSelectorStr(), format, request.Type)
+				continue
+			}
+
+			if err := storage.Persist(request, p, data); err != nil {
+				seclog.Errorf("couldn't persist [%s] to %s storage: %v", p.GetSelectorStr(), request.Type, err)
+			} else {
+				tags := []string{"format:" + request.Format.String(), "storage_type:" + request.Type.String(), fmt.Sprintf("compression:%v", request.Compression)}
+				if err := m.statsdClient.Count(metrics.MetricActivityDumpSizeInBytes, int64(data.Len()), tags, 1.0); err != nil {
+					seclog.Warnf("couldn't send %s metric: %v", metrics.MetricActivityDumpSizeInBytes, err)
+				}
+				if err := m.statsdClient.Count(metrics.MetricActivityDumpPersistedDumps, 1, tags, 1.0); err != nil {
+					seclog.Warnf("couldn't send %s metric: %v", metrics.MetricActivityDumpPersistedDumps, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// ListAllProfileStates list all profiles and their versions (debug purpose only)
+func (m *Manager) ListAllProfileStates() {
+	m.profilesLock.Lock()
+	defer m.profilesLock.Unlock()
+	for _, profile := range m.profiles {
+		profile.ListAllVersionStates()
+	}
+}
+
+// GetProfile returns a profile by its selector
+func (m *Manager) GetProfile(selector cgroupModel.WorkloadSelector) *profile.Profile {
+	m.profilesLock.Lock()
+	defer m.profilesLock.Unlock()
+
+	// check if this workload had a Security Profile
+	return m.profiles[selector]
+}
+
+// AddProfile adds a profile to the manager
+func (m *Manager) AddProfile(profile *profile.Profile) {
+	m.newProfiles <- profile
+}
```

Comment:
maybe add a comment to tell it's only used for test purposes?

Reply:
I added a testing.go file with 2d49296c2e74403069d4fe5c6781aa668f7ca980 that only contains manager methods used by functional tests